securedrop/crypto_util.py

```diff
@@ -225,6 +226,7 @@ def delete_reply_keypair(self, source_filesystem_id):
         # The subkeys keyword argument deletes both secret and public keys.
         temp_gpg.delete_keys(key, secret=True, subkeys=True)
 
+    @lru_cache(maxsize=500)
```

Comment:
so i know you're working on the other bits of this (cache warmup) so will hold off on review

i'm wondering: what do you think is a reasonable number of sources here to cache? this `lru_cache` is just storing a dict so maybe we could make this much larger, like 2048 (the [docs say](https://docs.python.org/3/library/functools.html#functools.lru_cache) performance is best when we use powers of 2)

Reply:
I went with 1000. I think that would be under 200kb, based on my crude estimates putting `filesystem_id` and `fingerprint` sized strings into `collections.OrderedDict`. 

I don't think the size needs to be a power of 2 with OrderedDict.

Reply:
afaik the LRU cache internally is using both a regular dict and a circular doubly linked list, I believe the latter is the reason for the powers of 2 suggestion in the docs

Reply:
Right, the latest iteration is no longer using `functools.lru_cache`.

Reply:
hmm why do we not want to use the upstream decorator?

Reply:
Doh, sorry, didn't update the PR description from the new commit message, which tried to explain that. Communication!